gustaf/spline/create.py

```diff
@@ -254,6 +255,289 @@ def revolved(
+    a : float
+      dimension of the rectangle in x-direction
+    b : float
+      dimension of the rectangle in y-direction
+    Returns
+    -------
+    gustaf.Bezier
+      Rectangluar Bezier spline with lengths of the sides a and b.
+    """
+    from gustaf.spline import Bezier
+
+    cps = np.zeros([4, 3])
+    cps[1, 0] = a
+    cps[2, 1] = b
+    cps[3, :2] = a, b
+
+    return Bezier(control_points=cps, degrees=[1, 1])
+
+
+def box(a, b, h):
+    """Volumetric spline with linear degree and dimensions a,b,h in
+    x,y,z-direction
+
+    Parameters
+    ----------
+    a : float
+      dimension of the box in x-direction
+    b : float
+      dimension of the box in y-direction
+    h : float
+      dimension of the box in z-direction
+
+    Returns
+    -------
+    gustaf.Bezier
+      Volumetric linear Bezier spline
+    """
+    return rectangle(a, b).create.extruded([0., 0., h])
+
+
+def disk(R, r0=0., angle=360., n_knot_spans=4):
+    """Surface spline describing a potentially hollow disk with quadratic
+    degree along curved dimension and linear along thickness.
+    The angle describes the returned part of the disk.
+
+    Parameters
+    ----------
+    R : float
+      Outer radius of the disk
+    r0 : float, optional
+      Inner radius of the disk, in case of hollow disk, by default 0.
+    angle : float, optional
+      Rotational angle, by default 360. describing a complete revolution
+    n_knot_spans : int, optional
+      Number of knot spans, by default 4
+
+    Returns
+    -------
+    gustaf.NURBS
+      Surface NURBS of degrees (1,2)
+    """
+
+    from gustaf.spline import NURBS
+
+    cps = np.array([[r0, 0., 0.], [R, 0., 0.]])
+    weights = np.ones([cps.shape[0]])
+    knots = np.repeat([0., 1.], 2)
+
+    return NURBS(
+            control_points=cps,
+            knot_vectors=[knots],
+            degrees=[1],
+            weights=weights
+    ).create.revolved(
+            axis=[0., 0., 1.],
+            center=np.zeros(3),
+            angle=angle,
+            n_knot_spans=n_knot_spans
+    )
+
+
+def torus(R, r, r0=0., angle=[360., 360.], n_knot_spans=[4, 4]):
+    """Creates a volumetric NURBS spline describing a torus with radius R
+    and cross-sectional radius r.
+
+    Parameters
+    ----------
+    R : float
+      Radius of the torus
+    r : float
+      Radius of the section of the torus
+    r0 : float, optional
+      Inner radius in case of hollow torus, by default 0.
+    angle : int, optional
+      Rotational angle, by default 360. describing a complete revolution
+    n_knot_spans : list, optional
+      Number of knot spans in 0: circular direction, 1: cross-sectional
+      direction, by default both equal 4
+
+    Returns
+    -------
+    gustaf.NURBS
+      Volumetric spline in the shape of a torus with degrees (1,2,2)
+    """
+
+    # Create a surface spline representing a disk and move it from the origin
+    d = disk(r, r0, n_knot_spans=n_knot_spans[1], angle=angle[1])
```

Comment:
Computationally I would prefer to have a tube-torus defined differrently from a full torrus(-segment). I.e., have a spline like this:
```
plate = gus.RationalBezier(
        degrees=[2,2],
        control_points=[
                                [0, 0], 
                                [0.5, -0.5],
                                [1, 0],
                                [-0.5, 0.5], 
                                [0.5, 0.5], 
                                [1 + 0.5, 0.5],
                                [0, 1],
                                [0.5, 1 + 0.5],
                                [1, 1],
                        ],
        weights=[1, isqrt2, 1, isqrt2, 1, isqrt2, 1, isqrt2, 1]
)
```
Be revolved to a torus